crate2nix/templates/nix/crate2nix/default.nix

```diff
@@ -111,9 +111,30 @@ rec {
           buildInputs = testInputs;
         } ''
           set -ex
-          cd ${crate.src}
+
+          export RUST_BACKTRACE=1
+
+          # recreate a file hierarchy as when running tests with cargo
+
+          # the source for test data
+          cp -r ${crate.src}/* .
+
+          # build outputs
+          testRoot=target/debug
+          mkdir -p $testRoot
+
+          # executables of the crate
+          cp -r ${crate}/bin/* $testRoot
+          chmod +w -R .
+
+          # test harness executables are suffixed with a hash
+          # this allows to prevent name collision with the main
+          # executables of the crate
+          hash=$(basename $out)
           for file in ${drv}/tests/*; do
-            $file $testCrateFlags 2>&1 | tee -a $out
+            f=$testRoot/$(basename $file)-$hash
+            cp $file $f
```

Comment:
So I guess that copying the binaries (above) is necessary so that they are found by the test framework that you are using in https://github.com/symphorien/nix-du.

But why copy the test executables themselves? Do they really check their own path rather than the current working directory? And if yes, wouldn't a symlink surfice? Or could you delete them after a test run?

Either way, the comment should make this clear.

Reply:
Yes, cli_test_dir uses `std::env::current_exe()`. So a symlink would resolve to the read only one.